test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,278 @@
+
+	tfIntraNode := fillTraceflow(pod0, pod1)
+	tfInterNode := fillTraceflow(pod0, pod2)
+
+	testcases := []testcase{
+		{
+			tf:            tfIntraNode,
+			expectedPhase: v1alpha1.Succeeded,
+			expectedResults: []v1alpha1.NodeResult{
+				{
+					Node: node1,
+					Observations: []v1alpha1.Observation{
+						{
+							Component: v1alpha1.SpoofGuard,
+							Action:    v1alpha1.Forwarded,
+						},
+						{
+							Component:     v1alpha1.NetworkPolicy,
+							ComponentInfo: "EgressRule",
+							Action:        v1alpha1.Forwarded,
+						},
+						{
+							Component:     v1alpha1.NetworkPolicy,
+							ComponentInfo: "IngressDefaultRule",
+							Action:        v1alpha1.Dropped,
+						},
+					},
+				},
+			},
+		},
+		{
+			tf:            tfInterNode,
+			expectedPhase: v1alpha1.Succeeded,
+			expectedResults: []v1alpha1.NodeResult{
+				{
+					Node: node1,
+					Observations: []v1alpha1.Observation{
+						{
+							Component: v1alpha1.SpoofGuard,
+							Action:    v1alpha1.Forwarded,
+						},
+						{
+							Component:     v1alpha1.NetworkPolicy,
+							ComponentInfo: "EgressRule",
+							Action:        v1alpha1.Forwarded,
+						},
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Output",
+							Action:        v1alpha1.Forwarded,
+						},
+					},
+				},
+				{
+					Node: node2,
+					Observations: []v1alpha1.Observation{
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Classification",
+							Action:        v1alpha1.Received,
+						},
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Output",
+							Action:        v1alpha1.Delivered,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	localConfig := rest.CopyConfig(data.kubeConfig)
+	client, err := versioned.NewForConfig(localConfig)
+	require.NoError(t, err)
+
+	// Setup 2 NetworkPolicies:
+	// 1. Allow all egress traffic.
+	// 2. Deny ingress traffic on pod with label antrea-e2e = pod1. So flow pod0 -> pod1 will be dropped.
+	var allowAllEgress *networkingv1.NetworkPolicy
+	if allowAllEgress, err = data.createNPAllowAllEgress("test-networkpolicy-allow-all-egress"); err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(allowAllEgress); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	var denyIngress *networkingv1.NetworkPolicy
+	if denyIngress, err = data.createNPDenyIngress("antrea-e2e", pod1, "test-networkpolicy-deny-ingress"); err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(denyIngress); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	// Create 2 traceflows:
+	// 1. pod0 -> pod1, intra node1.
+	// 2. pod0 -> pod2, inter node1 and node2.
+	client.OpsV1alpha1().Traceflows().Create(context.TODO(), tfIntraNode, v1.CreateOptions{})
```

Comment:
we don't check for an error here ? same comment for other `Create` calls.

Reply:
Updated.